projects/bms_carrier/src/cell_sense.c

```diff
@@ -29,6 +31,9 @@ static void prv_extract_cell_result(uint16_t *result_arr, size_t len, void *cont
     }
   }
 
+  // Balance cells if needed
+  passive_balance(result_arr, len, s_storage.afe);
```

Comment:
`s_storage.readings->voltages` should be used here I think, the assumption is the prior array could be modified during execution (hence the critical section)